apps/charterafrica/src/payload/utils/getPageUrlUsingPayload.js

```diff
@@ -0,0 +1,21 @@
+const getPageUrlUsingPayload = async (payload, slug) => {
+  const collection = "pages";
+  const options = {
+    where: {
+      slug: {
+        equals: slug,
+      },
+    },
+  };
+
+  const { docs } = await payload.find({
+    limit: 0,
+    ...options,
+    collection,
+  });
+
+  const breadcrumbs = docs[0]?.breadcrumbs || [];
+  return breadcrumbs[breadcrumbs.length - 1]?.url;
```

Comment:
We already have a [function](https://github.com/CodeForAfrica/ui/blob/55ee12c0a53855c3055fd788a80bae204722a26d/apps/charterafrica/src/payload/utils/formatPagePath.js#L1) to return page path/url given a page.